mangaki/mangaki/admin.py

```diff
@@ -355,6 +358,65 @@ def refresh_work(self, request, queryset):
 
     refresh_work.short_description = "Mettre à jour la fiche de l'anime (poster)"
 
+    @transaction.atomic
+    def change_title(self, request, queryset):
+        if request.POST.get('confirm'):  # Changing default title has been confirmed
+            new_titles = {}
+
+            for key in request.POST:
+                if key.isnumeric():
+                    work_id = int(key)
+                    if request.POST.get(key).isnumeric():
+                        title_id = int(request.POST.get(key))
+                        new_titles[work_id] = title_id
+
+            titles = WorkTitle.objects.filter(
+                pk__in=new_titles.values(), work__id__in=new_titles.keys()
+            ).values_list('title', 'work__title', 'work__id')
+
+            for new_title, current_title, work_id in titles:
+                if new_title != current_title:
+                    Work.objects.filter(pk=work_id).update(title=new_title)
+
+            self.message_user(request, 'Les titres ont bien été changés pour les œuvres sélectionnées.')
+            return None
+
+        work_titles = WorkTitle.objects.filter(work__in=queryset.values_list('pk', flat=True))
+        titles = {}
+
+        for work in queryset:
+            work_titles_for_work = work_titles.filter(work=work)
+            if len(work_titles_for_work) <= 0:
+                continue
+
+            infos = work_titles_for_work.values_list(
+                'pk', 'title', 'language', 'type').order_by('title').distinct('title')
+
+            titles[work.id] = {
+                info[0]: {
+                    'title': info[1],
+                    'language': info[2],
+                    'type': info[3] if info[1] != work.title else 'current'
+                } for info in infos
+            }
+
+        if len(titles) > 0:
```

Comment:
Another way to write it is: `if titles:` (i.e. if the `titles` array is not empty).